packages/api/src/repository.js

```diff
@@ -54,17 +55,33 @@ class Repository {
     return data.ticket;
   }
 
-  async performTicketIfAllowed (ticket, permission) {
-    if (ticket.action === 'delete-repo') {
+  async allowedToPerformTicket (action, permission) {
+    if (action === 'delete-repo') {
       return permission === 'admin';
     }
 
-    if (ticket.action === 'delete-test') {
+    if (action === 'delete-test') {
       return permission === 'admin' || permission === 'write';
     }
 
     return false;
-    // TODO actually perform the action
+  }
+
+  async performTicketIfAllowed (ticket, permission) {
+    const permitted = this.allowedToPerformTicket(ticket.action, permission);
+
+    if (permitted) {
+      if (ticket.action === 'delete-test') {
+        deleter.deleteTest(ticket.fullName, ticket.testName, client);
+      } else if (ticket.action === 'delete-repo') {
+        deleter.deleteRepo(client, ticket.fullName);
+      } else {
+        console.log('Invalid action');
```

Comment:
nit: might be worth making this `throw Error('invalid action')`, since this is really strange behavior and we'd like to bubble it.